### PR TITLE
9/sdk

### LINKED ISSUE #9: Use graasp sdk

### PR DESCRIPTION
**Pull request by pyphilia**
close #9

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,9 +6,16 @@
   "types": "dist/service-api.d.ts",
   "scripts": {
     "build": "tsc",
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "prepare": "yarn build",
-    "prepack": "yarn build"
+    "test": "echo \"Error: no test specified\" && exit 0",
```
Comment (dialexo) [inline package.json:9]: Should this be jest?

File: src/tasks/item-login-with-member-id-task.ts [hunk 2]
```diff
@@ -1,26 +1,48 @@
 // global
```
Comment (dialexo) [inline src/tasks/item-login-with-member-id-task.ts:1]: Should we remove this comment now that the imports sort is automated?
```suggestion

```

File: src/tasks/item-login-with-member-id-task.ts [hunk 3]
```diff
@@ -1,26 +1,48 @@
...
+  DatabaseTransactionHandler,
+  ItemMembershipService,
+  ItemService,
+  MemberService,
+  TaskStatus,
+} from '@graasp/sdk';
+
 import { ItemLoginService } from '../db-service';
 import { ItemLoginMemberExtra } from '../interfaces/item-login';
+// local
```
Comment (dialexo) [inline src/tasks/item-login-with-member-id-task.ts:13]: Should we remove this comment now that the imports sort is automated?
```suggestion

```

File: src/util/aux.ts [hunk 4]
```diff
@@ -1,15 +1,17 @@
 import bcrypt from 'bcrypt';
+
 import { ItemLoginSchema } from '../interfaces/item-login';
 
-const randomHexOf4 = () => (Math.random() * (1 << 16) | 0).toString(16).padStart(4, '0');
+const randomHexOf4 = () => ((Math.random() * (1 << 16)) | 0).toString(16).padStart(4, '0');
```
Comment (dialexo) [inline src/util/aux.ts:5]: This is probably fine for our purposes, though we may want to use e.g. https://www.npmjs.com/package/random-number-csprng instead. We could imagine an attack where a malicious user is aware of the time of the request of another specific user, and thus could deduce the identity of the pseudonymized user based on their activity.
See https://gist.github.com/joepie91/7105003c3b26e65efcea63f3db82dfba

File: src/util/graasp-item-login-error.ts [hunk 5]
```diff
@@ -1,79 +1,130 @@
...
-    this.code = code;
-    this.message = message;
-    this.statusCode = statusCode;
-    this.data = data;
-    this.origin = 'plugin';
-  }
-}
+import { PLUGIN_NAME } from './constants';
+
+export const GraaspError = ErrorFactory(PLUGIN_NAME);
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:7]:
```suggestion
export const GraaspItemLoginError = ErrorFactory(PLUGIN_NAME);
```

File: src/util/graasp-item-login-error.ts [hunk 6]
```diff
@@ -1,79 +1,130 @@
...
 
-export class ItemNotFound extends GraaspItemLoginError {
+export class ItemNotFound extends GraaspError {
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:9]:
```suggestion
export class ItemNotFound extends GraaspItemLoginError {
```

File: src/util/graasp-item-login-error.ts [hunk 7]
```diff
@@ -1,79 +1,130 @@
...
-    super({ code: 'GILERR001', statusCode: 404, message: 'Item not found' }, data);
+    super(
+      { code: 'GILERR001', statusCode: StatusCodes.NOT_FOUND, message: 'Item not found' },
+      data,
+    );
   }
 }
 
-export class MemberIdentifierNotFound extends GraaspItemLoginError {
+export class MemberIdentifierNotFound extends GraaspError {
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:18]:
```suggestion
export class MemberIdentifierNotFound extends GraaspItemLoginError {
```

File: src/util/graasp-item-login-error.ts [hunk 8]
```diff
@@ -1,79 +1,130 @@
...
+        statusCode: StatusCodes.NOT_FOUND,
+        message: 'Member identifier not found',
+      },
+      data,
+    );
   }
 }
 
-export class InvalidMember extends GraaspItemLoginError {
+export class InvalidMember extends GraaspError {
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:31]:
```suggestion
export class InvalidMember extends GraaspItemLoginError {
```

File: src/util/graasp-item-login-error.ts [hunk 9]
```diff
@@ -1,79 +1,130 @@
...
+        statusCode: StatusCodes.NOT_FOUND,
+        message: 'This member cannot be used to login to an item',
+      },
+      data,
+    );
   }
 }
 
-export class MissingItemLoginSchema extends GraaspItemLoginError {
+export class MissingItemLoginSchema extends GraaspError {
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:44]:
```suggestion
export class MissingItemLoginSchema extends GraaspItemLoginError {
```

File: src/util/graasp-item-login-error.ts [hunk 10]
```diff
@@ -1,79 +1,130 @@
...
+        statusCode: StatusCodes.INTERNAL_SERVER_ERROR,
+        message: 'Missing login schema',
+      },
+      data,
+    );
   }
 }
 
-export class MissingItemLoginTag extends GraaspItemLoginError {
+export class MissingItemLoginTag extends GraaspError {
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:57]:
```suggestion
export class MissingItemLoginTag extends GraaspItemLoginError {
```

File: src/util/graasp-item-login-error.ts [hunk 11]
```diff
@@ -1,79 +1,130 @@
...
+        statusCode: StatusCodes.BAD_REQUEST,
+        message: 'Item does not possess the required tag',
+      },
+      data,
+    );
   }
 }
 
-export class ValidMemberSession extends GraaspItemLoginError {
+export class ValidMemberSession extends GraaspError {
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:70]:
```suggestion
export class ValidMemberSession extends GraaspItemLoginError {
```

File: src/util/graasp-item-login-error.ts [hunk 12]
```diff
@@ -1,79 +1,130 @@
...
+        statusCode: StatusCodes.BAD_REQUEST,
+        message: 'Member with valid session trying to (re)login',
+      },
+      data,
+    );
   }
 }
 
-export class InvalidCredentials extends GraaspItemLoginError {
+export class InvalidCredentials extends GraaspError {
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:83]:
```suggestion
export class InvalidCredentials extends GraaspItemLoginError {
```

File: src/util/graasp-item-login-error.ts [hunk 13]
```diff
@@ -1,79 +1,130 @@
...
+        // eslint-disable-next-line quotes
+        message: "Provided credentials don't match member's",
+      },
+      data,
+    );
   }
 }
 
-export class MissingCredentialsForLoginSchema extends GraaspItemLoginError {
+export class MissingCredentialsForLoginSchema extends GraaspError {
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:97]:
```suggestion
export class MissingCredentialsForLoginSchema extends GraaspItemLoginError {
```

File: src/util/graasp-item-login-error.ts [hunk 14]
```diff
@@ -1,79 +1,130 @@
...
+        statusCode: StatusCodes.BAD_REQUEST,
+        message: 'Missing credentials for set login schema',
+      },
+      data,
+    );
   }
 }
 
-export class UnnecessaryCredentialsForLoginSchema extends GraaspItemLoginError {
+export class UnnecessaryCredentialsForLoginSchema extends GraaspError {
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:110]:
```suggestion
export class UnnecessaryCredentialsForLoginSchema extends GraaspItemLoginError {
```

File: src/util/graasp-item-login-error.ts [hunk 15]
```diff
@@ -1,79 +1,130 @@
...
+        statusCode: StatusCodes.BAD_REQUEST,
+        message: 'Unnecessary credentials for set login schema',
+      },
+      data,
+    );
   }
 }
 
-export class MemberCannotAdminItem extends GraaspItemLoginError {
+export class MemberCannotAdminItem extends GraaspError {
```
Comment (dialexo) [inline src/util/graasp-item-login-error.ts:123]:
```suggestion
export class MemberCannotAdminItem extends GraaspItemLoginError {
```

**Review by dialexo — CHANGES_REQUESTED** [review]
Thanks for this PR 🍘 I have one major security concern, and the `test` command should probably be set to jest. Otherwise, a few nitpicks regarding comments and naming 👍

File: package.json (hunk 1 above)
Comment (pyphilia) [reply package.json:9]: Yes but we don't have test yet.

**Pull request merged**